src/Symfony/Component/Validator/Constraints/BicValidator.php

```diff
@@ -81,5 +92,38 @@ public function validate($value, Constraint $constraint)
 
             return;
         }
+
+        // check against an IBAN
+        if ($path = $constraint->ibanPropertyPath) {
+            if (null !== $object = $this->context->getObject()) {
+                try {
+                    $iban = $this->getPropertyAccessor()->getValue($object, $path);
+                } catch (NoSuchPropertyException $e) {
+                    throw new ConstraintDefinitionException(sprintf('Invalid property path "%s" provided to "%s" constraint: %s', $path, \get_class($constraint), $e->getMessage()), 0, $e);
+                }
+            }
+        } else {
+            $iban = $constraint->iban;
+        }
+        if (!$iban) {
```

Comment:
shouldnt we prefer `nulll === $iban`  instead?

Reply:
@nicolas-grekas tell me whatever you prefer!

Reply:
I'm fine with `!$iban` personally: the logic below also skips e.g. the empty string or other falsy values.

Reply:
right. different question then :)

shouldnt we expect a violation for arbitrary ibans?

Reply:
you would get one by when adding a constraint on the iban field itself, isn't it enough?